src/lib/es2019.array.d.ts

```diff
@@ -17,26 +16,18 @@ interface ReadonlyArray<T> {
      * @param thisArg An object to which the this keyword can refer in the callback function. If
      * thisArg is omitted, undefined is used as the this value.
      */
-    flatMap<U, This = undefined> (
-        callback: (this: This, value: T, index: number, array: T[]) => U | ReadonlyArray<U>,
-        thisArg?: This

```

Comment:
An improvement in consistency isn't worth changing the built-in types, especially to make typing looser. I'd rever thist.

Reply:
But you know, TypeScript once revert #12784 in #16223 due to performance issue. If it is not the case nowadays, why not redoing #12784?

Reply:
It's only worth changing the types if there's a performance problem or something else major. Any change, even a fix, will break some code somewhere, and I want to avoid that in this case.